facebook/facebook-ios-sdk
# SPM Support

Thanks for proposing a pull request!

To help us review the request, please complete the following:


## Pull Request Details

Added a `Package.swift` manifest. 
Updated imports to be non-modular for non-cocoapod setups. 
Updated dynamic framework loader to compile with arc. - This should be checked. Not 100% sure I did not cause a regression.
Added a directory of symlinks to public header files - this was to get around a limitation in the SPM api where they will only accepts a path to header files (forces them to be located in the same directory) and not a list of header files.

Additional hacks/workarounds are commented in the changes.

## Test Plan

Test Plan:

1. Clone the [fork](https://github.com/joesus/facebook-ios-sdk) and checkout the branch 

2. Open the package itself with >= Xcode 11 by opening `Package.swift` (as opposed to the usual project file or workspace) and verify that it builds

3. Open a new project with >= Xcode 11 and 'Add a New Swift Package'. For the Package Repository field put the path to your checkout. Ex: `file:///Users/$whoami/facebook-ios-sdk` then specify the branch as this one, `spm-corekit`. This sample project should build and you should be able to consume the kits with `import FBSDKCoreKit`, `import FBSDKLoginKit` and `import FBSDKShareKit`. You should not need to import `FBSDKCoreKit` to access core symbols such as `Settings` if you've already imported one of the other kits.

Additionally check: 

**Xcode:**
Can build all schemes as expected
`BuildAllKits`
`FBSDKTVOSKit-Universal`
`FBSDKTVOSKit-Dynamic`
`SwiftKits`

**CocoaPods:**
`bundle exec pod lib lint FBSDKCoreKit.podspec`
`bundle exec pod lib lint FBSDKLoginKit.podspec --include-podspecs=FBSDKCoreKit.podspec --allow-warnings`
`bundle exec pod lib lint FBSDKShareKit.podspec --include-podspecs=FBSDKCoreKit.podspec --allow-warnings`
